tests/test_sshfsmount.cpp

```diff
@@ -59,10 +67,52 @@ struct SshfsMount : public mp::test::SftpServerTest
         return request_exec;
     }
 
+    auto make_exec_to_check_commands(const command_vector& commands, std::string::size_type& remaining, command_vector::const_iterator& next_expected_cmd, std::string& output, bool& invoked)
+    {
+        auto request_exec = [&commands, &remaining, &next_expected_cmd, &output, &invoked](ssh_channel, const char* raw_cmd) {
+            std::string cmd{raw_cmd};
+            mp::utils::trim_end(cmd);
+
+            if (next_expected_cmd != commands.end())
+            {
+                // Check if the first of the remaining commands is executed first.
+                if (cmd == next_expected_cmd->first)
+                {
+                    output = next_expected_cmd->second;
+                    remaining = output.size();
+                    invoked = true;
+                    ++next_expected_cmd;
+                }
+                else
+                {
+                    // If not, then we have to check the rest of the remaining
+                    // commands. If the executed command is there, it means that the
+                    // executed commands do not follow the order of our expected
+                    // command vector.
+                    for (auto it2 = std::next(next_expected_cmd); it2 != commands.end(); ++it2)
```

Comment:
It's the only iterator, just `it` would be enough?

Reply:
Sure, the name remains from a previous implementation. Changed.